### PR TITLE
Feature/Adds Inbound SMS Parser

### PR DESCRIPTION
**Pull request by CJButton**
## Description:

Parser takes in a byte slice `[]byte` and returns a WebHook struct.

*** Notes ***

Expected usage is as follows: 
```
r, _ := ioutil.ReadAll(c.Request.Body)
w := ParseJSONWebHook(r)
```

The parser function is now more agnostic about who uses it. For example, users could use the built-in `net/http` package and parse it by doing `r, _ := ioutil.ReadAll(req.Body)`.

Another option is to use the decoder.
`w := json.NewDecoder(c.Request.Body)`

But that would require changes to the Parser function as well.
```
func ParseJSONWebHook(d *json.Decoder) WebHook {
	var w WebHook
	d.Decode(&w)
	return w
}
```

Finally, this has the expectation that the request method will be `JSON(POST)`. However, in the SMS console under `リクエストメソッド ` (request method), there are three options: `GET`, `POST`, `JSON(POST)`. If we wish to parse the other requests, then we will need a different function. I believe default selection is `GET` or `POST`. Perhaps the support team can change the request method upon creating the user.

## Added

* Testify and basic test for the parser.
* More fields to the WebHook to account for wider range of requests.

## References:

https://comvex.myjetbrains.com/youtrack/agiles/107-0/108-168?issue=DGM2-2870

### CODE REVIEW

**Review by cliff-comvex — APPROVED** [review]
Let's wait for others to review as I am not confident about my review 😅

File: utilities/parse_json_webhook.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+	"encoding/json"
+
+	"github.com/comvex-jp/mediasms-go/models"
+)
+
+// ParseJSONWebHook returns a WebHook struct
+func ParseJSONWebHook(body []byte) models.WebHook {
+	var w models.WebHook
+	json.Unmarshal(body, &w)
+	return w
```
Comment (anderson-comvex) [inline utilities/parse_json_webhook.go]:
```suggestion
	var w models.WebHook
	
	json.Unmarshal(body, &w)
	
	return w
```

File: utilities/parse_json_webhook_test.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	"originalurl4":      "",
+	"type":              "sms",
+	"returnsms":         "1",
+	"waitreturnsms":     "1",
+	"returnsmsdatetime": "2020年09月01日 10:00:00",
+	"replyid":           "1",
+	"senderid":          "222211110000"}
+
+func TestParseJSONWebHook(t *testing.T) {
+
```
Comment (anderson-comvex) [inline utilities/parse_json_webhook_test.go]: Can we remove this empty line?

**Review by anderson-comvex — CHANGES_REQUESTED** [review]

**Review by anderson-comvex — APPROVED** [review]

File: utilities/parse_json_webhook_test.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	"clicks3":           "0",
+	"clicks4":           "0",
+	"originalurl":       "http://google.com",
+	"originalurl2":      "",
+	"originalurl3":      "",
+	"originalurl4":      "",
+	"type":              "sms",
+	"returnsms":         "1",
+	"waitreturnsms":     "1",
+	"returnsmsdatetime": "2020年09月01日 10:00:00",
```
Comment (rdehaas-comvex) [inline utilities/parse_json_webhook_test.go:26]: Wonder if the parser should be responsible for changing this into something more generic (e.g. `2020-09-01 10:00:00`)?

Comment (CJButton) [reply utilities/parse_json_webhook_test.go:26]: @rdehaas-comvex Do you think we should translate the Status to English, or should we let that be done on the SMS-API side? Potentially, the user would actually want it in Japanese I guess. Although I suspect we will be the only users of this library.

Comment (rdehaas-comvex) [reply utilities/parse_json_webhook_test.go:26]: Good point, `status` could possibly stay the same 🤔 
Unless we want the whole library to be 100% English for consistency.

**Review by rdehaas-comvex — CHANGES_REQUESTED** [review]

File: utilities/parse_json_webhook_test.go (hunk 3 above)
Comment (rdehaas-comvex) [reply utilities/parse_json_webhook_test.go:26]: Good point, `status` could possibly stay the same 🤔 
Unless we want the whole library to be 100% English for consistency.

Comment (cliff-comvex) [reply utilities/parse_json_webhook_test.go:26]: I think it's better we make the library 100% english including status

**Review by cliff-comvex — CHANGES_REQUESTED** [review]
I think it's better we make the library 100% english including status?

**Comment by CJButton** [comment]
@cliff-comvex Alright. We will go 100% English then.

**Review by cliff-comvex — APPROVED** [review]

**Comment by cliff-comvex** [comment]
This is okay to merge

**Pull request merged**